indra/newview/llfloaterinventorysettings.cpp

```diff
@@ -28,17 +28,38 @@
 
 #include "llfloaterinventorysettings.h"
 
+#include "llcolorswatch.h"
+
 LLFloaterInventorySettings::LLFloaterInventorySettings(const LLSD& key)
   : LLFloater(key)
 {
+    mCommitCallbackRegistrar.add("ScriptPref.applyUIColor", boost::bind(&LLFloaterInventorySettings::applyUIColor, this, _1, _2));
+    mCommitCallbackRegistrar.add("ScriptPref.getUIColor", boost::bind(&LLFloaterInventorySettings::getUIColor, this, _1, _2));
 }
 
 LLFloaterInventorySettings::~LLFloaterInventorySettings()
 {}
 
 BOOL LLFloaterInventorySettings::postBuild()
 {
-    getChild<LLButton>("ok_btn")->setCommitCallback(boost::bind(&LLFloater::closeFloater, this, false));
+    getChild<LLUICtrl>("favorites_color")->setCommitCallback(boost::bind(&LLFloaterInventorySettings::updateColorSwatch, this));
     return TRUE;
 }
 
+void LLFloaterInventorySettings::updateColorSwatch()
+{
+    bool val = getChild<LLUICtrl>("favorites_color")->getEnabled();
+    getChild<LLUICtrl>("favorites_color")->setEnabled(val);
```

Comment:
Not sure what happens here.

Reply:
It should have been getChild<LLUICtrl>("favorites_swatch")->setEnabled(val);